src/ImageSharp/ColorSpaces/Conversion/Implementation/Icc/Calculators/ClutCalculator.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Numerics;
+using SixLabors.ImageSharp.MetaData.Profiles.Icc;
+
+namespace SixLabors.ImageSharp.ColorSpaces.Conversion.Implementation.Icc.Calculators
+{
+    internal class ClutCalculator : IVector4Calculator
+    {
+        private int inputCount;
+        private int outputCount;
+        private float[][] lut;
```

Comment:
Is this array always jagged?

Reply:
It actually doesn't have to be, the second array is always the same length. In the Interpolate method it's currently useful though because I can just take the array reference instead of copying the values.
Do you think it would be better to have it in a single memory block?

Reply:
We have a helper `Fast2DArray<T>` which might be useful here for n-length 2D objects, it's faster than the jagged array. Though if we template the interpolation we should look at custom structs for each known 2D grid since they'd be much faster.